client/src/main/scala/skuber/PersistentVolumeClaim.scala

```diff
@@ -1,47 +1,59 @@
 package skuber
 
-import java.util.Date
-
-import Volume._
+import skuber.annotation.MatchExpression
 
 /**
- * @author David O'Riordan
- */
+  * @author David O'Riordan
+  *
+  */
+
 case class PersistentVolumeClaim(
-    val kind: String ="PersistentVolumeClaim",
+    val kind: String = "PersistentVolumeClaim",
     override val apiVersion: String = v1,
     val metadata: ObjectMeta = ObjectMeta(),
     spec: Option[PersistentVolumeClaim.Spec] = None,
     status: Option[PersistentVolumeClaim.Status] = None)
-      extends ObjectResource {
-  
-  def withResourceVersion(version: String) = this.copy(metadata = metadata.copy(resourceVersion=version))
+  extends ObjectResource {
+
+  def withResourceVersion(version: String) = this.copy(metadata = metadata.copy(resourceVersion = version))
 
 }
-   
+
+case class Selector(matchLabels: Option[Map[String, String]] = None, matchExpressions: Option[List[MatchExpression]] = None)
+
 object PersistentVolumeClaim {
 
-  val specification=CoreResourceSpecification(
+  object VolumeMode extends Enumeration {
+    type VolumeMode = Value
+    val Filesystem, BlockVolume = Value
+  }
+
+  val specification = CoreResourceSpecification(
     scope = ResourceSpecification.Scope.Namespaced,
     names = ResourceSpecification.Names(
-      plural="persistentvolumeclaims",
-      singular="persistentvolumeclaim",
-      kind="PersistentVolumeClaim",
-      shortNames=List("pvc")
+      plural = "persistentvolumeclaims",
+      singular = "persistentvolumeclaim",
+      kind = "PersistentVolumeClaim",
+      shortNames = List("pvc")
     )
   )
-  implicit val pvcDef = new ResourceDefinition[PersistentVolumeClaim] { def spec=specification }
-  implicit val pvcListDef = new ResourceDefinition[PersistentVolumeClaimList] { def spec=specification }
+  implicit val pvcDef = new ResourceDefinition[PersistentVolumeClaim] { val spec = specification }
+  implicit val pvcListDef = new ResourceDefinition[PersistentVolumeClaimList] { val spec = specification }
 
   import PersistentVolume.AccessMode
   case class Spec(
-      accessModes: List[AccessMode.AccessMode] = Nil,
-      resources: Option[Resource.Requirements] = None,
-      volumeName: String="",
-      storageClassName: Option[String] = None)
-      
-  import PersistentVolume.Phase    
+    accessModes: List[AccessMode.AccessMode] = Nil,
+    volumeMode: VolumeMode.VolumeMode,
```

Comment:
I believe this field is optional, and doesn't exist in older versions of Kubernetes. Can we make this an Option[VolumeMode] type with a default of None, and (for best compatibility) place it after the existing fields in the case class please?

Reply:
@doriordan good point! I've updated the PR following your recommendation.

Reply:
@doriordan WDYT? Could we merge this? We would really like to use the new PVC implementation in our application.